### PR TITLE
Newsletter Api Implementation

### PR DESCRIPTION
**Pull request by Solomonuche**
This PR contains the implementation of **`[POST] /api/v1/pages/newsletter`** **[UNPROTECTED Endpoint]** 

## Description
Develop a backend endpoint to handle requests for newsletter subscriptions. The endpoint will accept an email address, validate and sanitize it, and then handle both successful and failed validations appropriately. If the email passes validation, it will be stored in the database and a 'success' response will be sent to the client. If validation fails, an error message will be returned.

## Related Issue (Link to issue ticket)
This is an implementation of an open issue [Link](https://github.com/hngprojects/hng_boilerplate_python_fastapi_web/issues/46)

## changes made
I added **`email_validator`** and **`pytest-mock`** dependency to the requirements.txt file. pls pip install the requirement file in order to have everything need to run the code successfully
- email_validator is required by pydantic to have email validation
- pytest-mock is requirement for mocking database set-up during testing

## How Has This Been Tested?

This implementation has been thoroughly tested with fastapi openapi interface and postman

## Screenshots (if appropriate - Postman, etc):​
[Successful response](https://github.com/user-attachments/assets/a76712bb-2834-4b9a-aa83-05485bc05029)

[Error response](https://github.com/user-attachments/assets/bc05ec85-4631-47a3-a777-c048426e09f6)

## Checklist:

- [x ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x ] I have read the **CONTRIBUTING** document.
- [x ] I have added tests to cover my changes.
- [x ] All new and existing tests passed.

### CODE REVIEW

File: api/v1/routes/newsletter_router.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    )
+from fastapi.responses import JSONResponse
+from sqlalchemy.orm import Session
+from api.v1.models.newsletter import NEWSLETTER
+from api.v1.schemas.newsletter_schema import EMAILSCHEMA
+from api.db.database import get_db, Base, engine
+
+
+Base.metadata.create_all(bind=engine)
+print('creating table')
```
Comment (bolexs) [inline api/v1/routes/newsletter_router.py:16]: Can this be removed

**Review by bolexs — CHANGES_REQUESTED** [review]
Looks great, @Dev-wonderful @zxenonx check it out

**Comment by bolexs** [comment]
> Looks great, @Dev-wonderful @zxenonx check it out 

@zxenonx i Will merging this to dev

File: api/v1/routes/newsletter_router.py (hunk 1 above)
Comment (Solomonuche) [reply api/v1/routes/newsletter_router.py]: @bolexs I have removed the unnecessary Print statement used during development. Thank you for spotting that.

**Review by bolexs — APPROVED** [review]

**Pull request merged**